pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/column/IntermediateIndexContainer.java

```diff
@@ -61,9 +61,10 @@ public IntermediateIndexContainer(FieldSpec fieldSpec, @Nullable PartitionFuncti
   }
 
   public DataSource toDataSource(int numDocsIndexed) {
+    // TODO(Vivek): Change to correct value of maxRowLengthInBytes
     return new MutableDataSource(_fieldSpec, numDocsIndexed, _numValuesInfo._numValues,
         _numValuesInfo._maxNumValuesPerMVEntry, _dictionary.length(), _partitionFunction, _partitions, _minValue,
-        _maxValue, _forwardIndex, _dictionary, null, null, null, null, null, null, null, null);
+        _maxValue, _forwardIndex, _dictionary, null, null, null, null, null, null, null, null, -1);
```

Comment:
I'm not sure where IntermediateSegment is being used. Based on the other parameter, I assume using -1 should be fine! In any case, please remove the TODO.

Reply:
This class can actually be removed. It was introduced as an effort to unify the realtime and offline segment creation, but seems like the project is aborted.